File: src/main/java/com/marand/auditor/AuditDetailsReceiver.java
package com.marand.auditor;

import java.util.concurrent.atomic.AtomicInteger;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.jms.annotation.JmsListener;
import org.springframework.stereotype.Component;


/**
 * @author Nejc Korasa
 */

@Component
public class AuditDetailsReceiver
{
  private static final Logger LOG = LoggerFactory.getLogger(AuditDetailsReceiver.class);

  @JmsListener(destination = "auditQueue", containerFactory = "myFactory")
  public void receiveMessage(final AuditInfo auditInfo)
  {
    LOG.info("Received [" + auditInfo + "]");
  }
}

File: src/main/kotlin/com/marand/auditor/AuditDetailsReceiver.kt
package com.marand.auditor

import com.marand.auditor.db.AuditInfoRepository
import com.marand.auditor.dto.AuditInfo
import org.slf4j.LoggerFactory
import org.springframework.jms.annotation.JmsListener
import org.springframework.stereotype.Component


/**
 * @author Nejc Korasa
 */

@Component
class AuditDetailsReceiver(private val auditInfoRepository: AuditInfoRepository) {

    companion object {
        private val LOG = LoggerFactory.getLogger(AuditDetailsReceiver::class.java)
    }

    @JmsListener(destination = "auditQueue", containerFactory = "myFactory")
    fun receiveMessage(auditInfo: AuditInfo) {

        LOG.info("Received [$auditInfo]")
        auditInfoRepository.save(AuditInfoMapper.fromDto(auditInfo))
    }
}
